packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2253,8 +2262,25 @@ class InputDecoration {
   /// null.
   ///
   /// The semantic label can be replaced by providing a [semanticCounterText].
+  ///
+  /// If set to empty string and [counter] isn't specified, will display nothing
+  /// in the place of the counter.
   final String counterText;
 
+  /// Optional function to generate a custom counter widget, called with
```

Comment:
We should call this buildCounter, to distinguish it from the widget valued parameters. 

Even though only 2 parameters are needed at the moment, it would probably be safest to use @required named parameters instead. That way, more parameters can be added in the future.